### ISSUE TITLE
Interruptions in state update

### ISSUE THREAD

**Issue by ThatMajd** [HomeSpan/HomeSpan#616]
Hello,
I wrote code for a water leak sensor, and I get periods (every 5 minutes or so) of the esp not reporting its state (the `L2` logs don't include  `{"characteristics":[{"iid":5,"value":0,"aid":2}]}
-------- SENT ENCRYPTED! --------`) and then it goes back to reporting normally
During this downtime typing `s` in the CLI shows me that all the connections are disconnected, and my code seems to be working since the `LOG1("Leak Update: ");` is showing up correctly in the Serial Monitor.
This is my code, and thanks in advance!
## Waterleak.ino
```
#include "HomeSpan.h"
#include "DEV_Sensors.h"

void setup() {     
 
  Serial.begin(115200);
  
  homeSpan.setPairingCode("11122333");
  homeSpan.begin();
  

  homeSpan.begin(Category::Bridges,"Sensor Hub");

  new SpanAccessory();  
    new Service::AccessoryInformation();
      new Characteristic::Identify(); 
      
  new SpanAccessory();
    new Service::AccessoryInformation();
      new Characteristic::Identify();
      new Characteristic::Name("Bucket 1");
    new DEV_LeakSensor();
    
} 

void loop(){

 homeSpan.poll(); 

} // end of loop()
```
## DEV_Sensor.h
```
struct DEV_LeakSensor : Service::LeakSensor {

  SpanCharacteristic *leak;
  int pin = 34;

  DEV_LeakSensor() : Service::LeakSensor(){
    
    leak = new Characteristic::LeakDetected(0);
    leak->setRange(0,1);
    pinMode(pin, INPUT);

  } // end constructor



  void loop(){

    if(leak->timeVal()>10000){
      bool t=leak->getVal<bool>();
      t = digitalRead(pin);
      
      leak->setVal(t);
      
      LOG1("Leak Update: ");
      LOG1(t);
      LOG1("\n");            
    }
    
  } // loop
  
};
```

**Comment by ThatMajd** [comment]
Yes

**Comment by HomeSpan** [comment]
Have you updated to the new HomeKit architecture as well?

**Comment by ThatMajd** [comment]
No, Is it necessary?

**Comment by HomeSpan** [comment]
I was having similar issues, which were resolved after I updated to the new architecture.  I believe the issues are a result of Apple making lots of continuous changes to HomeKit from iOS 14 through iOS 16 that led to substantive degradations.  

By using the HomeSpan log file I was able to track how the architecture changed, how notifications are propagated, and which clients connect to each device.  Originally, a device would need to maintain 5-10 connections, but now there is generally only one and rarely two.  

The rollout of the new architecture was difficult as other things broke (and Apple had pull the update) but it's been re-released and hopefully now works as expected. 

Can't say for certain whether this will solve the issue, but I found the new architecture is much more stable.  

That said, I also switched mesh networks from Linksys Velop to Netgear Orbi during this same period of time, and that has also led to a lot of improvements in WiFi connectivity, which also effects HomeKit.
